### PR TITLE
📊 Fix `clip_ratio` logging and better document logged values

### LINKED ISSUE #3032: The loss metircs in GRPO might be wrong
```python
is_clipped = (per_token_loss1 < per_token_loss2).float()
clip_ratio = (is_clipped * completion_mask).sum() / completion_mask.sum()
self._metrics[mode]["clip_ratio"].append(self.accelerator.gather_for_metrics(clip_ratio).mean().item())
return loss
```
Above code calculates **_is_clipped_** which determines whether the importance sampling ratio is clipped.
However,if the **_coef_1_** is larger than **_1+epsilon_**, this is also a situation which indicates **_is_clipped_=True**.

I guess that the computational method about **_is_clipped_** might be wrong.

### LINKED ISSUE #3144: What‘s the meaning of clip_ratio in GRPO Trainer?
I am trying to understand the rationale behind the calculation of `clip_ratio` in the following code snippet:
```python
coef_1 = torch.exp(per_token_logps - old_per_token_logps)
coef_2 = torch.clamp(coef_1, 1 - self.epsilon_low, 1 + self.epsilon_high)
per_token_loss1 = coef_1 * advantages.unsqueeze(1)
per_token_loss2 = coef_2 * advantages.unsqueeze(1)
per_token_loss = -torch.min(per_token_loss1, per_token_loss2)

is_clipped = (per_token_loss1 < per_token_loss2).float()
clip_ratio = (is_clipped * completion_mask).sum() / completion_mask.sum()
self._metrics[mode]["clip_ratio"].append(self.accelerator.gather_for_metrics(clip_ratio).mean().item())
```
If `clip_ratio` is intended to indicate how frequently the policy updates are constrained to prevent large changes, shouldn't the `is_clipped` be:
```python
is_clipped = (per_token_loss1 > per_token_loss2).float()
```
since we are using `torch.min(per_token_loss1, per_token_loss2)`.

I would appreciate any insights or clarification on this matter. Thank you!

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Fixes #3032
Fixes #3144

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
Screenshot 2025-03-23 at 12 29 20

**Review by kashif — APPROVED** [review]
